src/plugins/maps_ems/public/plugin.ts

```diff
@@ -6,45 +6,58 @@
  * Side Public License, v 1.
  */
 
-// @ts-ignore
 import { CoreSetup, CoreStart, Plugin, PluginInitializerContext } from 'kibana/public';
-// @ts-ignore
-import { setKibanaVersion, setMapsEmsConfig } from './kibana_services';
-// @ts-ignore
-import { MapsEmsPluginSetup, MapsEmsPluginStart } from './index';
-import type { MapsEmsConfig } from '../config';
-import { getServiceSettings } from './lazy_load_bundle/get_service_settings';
-
-/**
- * These are the interfaces with your public contracts. You should export these
- * for other plugins to use in _their_ `SetupDeps`/`StartDeps` interfaces.
- * @public
- */
-
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface MapsEmsStartDependencies {}
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface MapsEmsSetupDependencies {}
+import {
+  setKibanaVersion,
+  setLicensingPluginStart,
+  setMapConfig,
+  getIsEnterprisePlus,
+} from './kibana_services';
+import type { MapsEmsPluginPublicSetup, MapsEmsPluginPublicStart } from './index';
+import type { MapConfig } from '../config';
+import { createEMSSettings } from '../common/ems_settings';
+import type {
+  LicensingPluginSetup,
+  LicensingPluginStart,
+} from '../../../../x-pack/plugins/licensing/public';
+import { createEMSClientLazy } from './lazy_load_bundle';
+
+interface MapsEmsStartPublicDependencies {
+  licensing?: LicensingPluginStart;
+}
+interface MapsEmsSetupPublicDependencies {
+  licensing?: LicensingPluginSetup;
+}
 
-export class MapsEmsPlugin implements Plugin<MapsEmsPluginSetup, MapsEmsPluginStart> {
-  readonly _initializerContext: PluginInitializerContext<MapsEmsConfig>;
+export class MapsEmsPlugin implements Plugin<MapsEmsPluginPublicSetup, MapsEmsPluginPublicStart> {
+  readonly _initializerContext: PluginInitializerContext<MapConfig>;
 
-  constructor(initializerContext: PluginInitializerContext<MapsEmsConfig>) {
+  constructor(initializerContext: PluginInitializerContext<MapConfig>) {
     this._initializerContext = initializerContext;
   }
 
-  public setup(core: CoreSetup, plugins: MapsEmsSetupDependencies) {
-    const config = this._initializerContext.config.get<MapsEmsConfig>();
+  public setup(core: CoreSetup, plugins: MapsEmsSetupPublicDependencies) {
+    const mapConfig = this._initializerContext.config.get<MapConfig>();
     const kibanaVersion = this._initializerContext.env.packageInfo.version;
 
     setKibanaVersion(kibanaVersion);
-    setMapsEmsConfig(config);
+    setMapConfig(mapConfig);
 
     return {
-      getServiceSettings,
-      config,
+      config: mapConfig,
+      createEMSSettings: () => {
```

Comment:
getIsEnterprisePlus is not initialized in setup, instead, it's initialized in `start`. I do not think that getIsEnterprisePlus should be used in setup unless setLicensingPluginStart is also initialed in setup.